quotefault/__init__.py

```diff
@@ -28,69 +43,39 @@
                           issuer=app.config['OIDC_ISSUER'],
                           client_registration_info=app.config['OIDC_CLIENT_CONFIG'])
 
-# Create CSHLDAP connection
-_ldap = CSHLDAP(app.config["LDAP_BIND_DN"],
-                app.config["LDAP_BIND_PW"])
-
 app.secret_key = 'submission'  # allows message flashing, var not actually used
 
-from .ldap import get_all_members, ldap_get_member
-from .mail import send_quote_notification_email
-
-
-# create the quote table with all relevant columns
-class Quote(db.Model):
-    id = db.Column(db.Integer, primary_key=True, nullable=False)
-    submitter = db.Column(db.String(80), nullable=False)
-    quote = db.Column(db.String(200), unique=True, nullable=False)
-    speaker = db.Column(db.String(50), nullable=False)
-    quote_time = db.Column(db.DateTime, nullable=False)
-
-    # initialize a row for the Quote table
-    def __init__(self, submitter, quote, speaker):
-        self.quote_time = datetime.now()
-        self.submitter = submitter
-        self.quote = quote
-        self.speaker = speaker
-
-
-class Vote(db.Model):
-    id = db.Column(db.Integer, primary_key=True, nullable=False)
-    quote_id = db.Column(db.ForeignKey("quote.id"))
-    voter = db.Column(db.String(200), nullable=False)
-    direction = db.Column(db.Integer, nullable=False)
-    updated_time = db.Column(db.DateTime, nullable=False)
-
-    quote = db.relationship(Quote)
-    test = db.UniqueConstraint("quote_id", "voter")
-
-    # initialize a row for the Vote table
-    def __init__(self, quote_id, voter, direction):
-        self.quote_id = quote_id
-        self.voter = voter
-        self.direction = direction
-        self.updated_time = datetime.now()
-
+from .ldap import get_all_members, is_member_of_group
+from .mail import send_quote_notification_email, send_report_email
 
 def get_metadata():
+    """
+    Provide metadata to page
+    UUID, UID, Git Version, Plug, and admin permissions obtained
+    """
     uuid = str(session["userinfo"].get("sub", ""))
     uid = str(session["userinfo"].get("preferred_username", ""))
-    version = subprocess.check_output(['git', 'rev-parse', '--short', 'HEAD']).decode('utf-8').rstrip()
+    version = subprocess.check_output(
+            ['git', 'rev-parse', '--short', 'HEAD']
+        ).decode('utf-8').rstrip()
     plug = request.cookies.get('plug')
     metadata = {
         "uuid": uuid,
         "uid": uid,
         "version": version,
-        "plug": plug
+        "plug": plug,
+        "is_admin" : is_member_of_group(uid,'eboard') or is_member_of_group(uid,'rtp')
     }
     return metadata
 
-
-# run the main page by creating the table(s) in the CSH serverspace and rendering the mainpage template
+# run the main page by creating the table(s)
+# in the CSH serverspace and rendering the mainpage template

```

Comment:
this function no longer crates tables

Reply:
The new version doesn't make sense @jabbate19. What does `in the CSH serverspace and rendering the mainpage template` mean?